### PR TITLE
Laurence/insert msg and convo

### PR DESCRIPTION
**Pull request by laurencecnerual**
With no convo ID given a new one is made for message association and finally sent to the frontend
image

When convo ID is given, messages are associated with it and its updated_at is updated (from conversation table in db)
image

Example of a continued conversation (from message table in db)
image

KNOWN ISSUE - chatgpt is called a second time to receive a title for the conversation, which is not ideal, and also, the title currently received is too vague/general... ("Understanding coding concepts.", etc.)

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -27,22 +27,87 @@ app.get('/', (req, res) => {
...
   try {
     const openaiResponse = await openaiRequest(message);
-    res.json({ response: openaiResponse });
+
+    let associatedConversationID;
+    let continuedConversation;
+    //if no conversation ID present, add a new conversation and use its ID - otherwise use the ID given
+    if (!conversation_id) {
+      continuedConversation = false;
+      const openaiSummary = await openaiRequest("Summarize the specific topic of the question that was asked in 5 words or less.");
```
Comment (vicentbe10) [inline src/server.js:48]: Here you could call my recent PR from vicente/add-tittlle-generator to get the title

File: src/server.js [hunk 2]
```diff
@@ -27,22 +27,87 @@ app.get('/', (req, res) => {
...
+        conversation_id: associatedConversationID,
+        author: chatUsername,
+        content: message
+      };
+
+      //Preparing the message record for the answer received from ChatGPT
+      const gptAnswer = {
+        chat_user_id: user_id,
+        conversation_id: associatedConversationID,
+        author: "ChatGPT", //Is this text ok?
```
Comment (vicentbe10) [inline src/server.js:78]: It seems that there are 3 roles in the OpenAI:
'user', 'assistant' and 'system'
Probably better to stick to those.

**Review by vicentbe10 — COMMENTED** [review]
A couple of comments to see what do you think before approving. Thanks

**Comment by laurencecnerual** [comment]
Please approve and I will work on the new method of generating the conversation title in my next PR

**Pull request merged**